src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -13731,6 +13810,73 @@ def _window_agg(
                     for quoted_identifier in frame.data_column_snowflake_quoted_identifiers
                 }
             ).frame
+        elif agg_func == "corr":
+            if window_func == WindowFunction.ROLLING and not isinstance(window, int):
+                ErrorMessage.not_implemented(
+                    f"window {window} must be an integer value."
+                )
+            if window != min_periods:
+                ErrorMessage.not_implemented(
+                    f"min_periods {min_periods} must be == window {window}"
+                )
+            assert window == min_periods
+            other = agg_kwargs.get("other", None)
+            other_qc = other._query_compiler
+            result_frame, result_column_mapper = join_utils.align(
+                left=frame,
+                right=other_qc._modin_frame,
+                left_on=frame.index_column_snowflake_quoted_identifiers,
+                right_on=other_qc._modin_frame.index_column_snowflake_quoted_identifiers,
+            )
+
+            # columns that exist in both dfs
+            matching_col_label_dict = {}
+            for i in range(len(frame.data_column_pandas_labels)):
+                frame_label = frame.data_column_pandas_labels[i]
+                frame_identifier = frame.data_column_snowflake_quoted_identifiers[i]
+                other_frame_identifier = (
+                    other_qc._modin_frame.data_column_snowflake_quoted_identifiers[i]
+                )
+                if frame_label in other_qc._modin_frame.data_column_pandas_labels:
+                    matching_col_label_dict[frame_label] = [
+                        frame_identifier,
+                        other_frame_identifier,
+                    ]
+
+            corr_result = result_frame
+            # columns that do not exist in both dfs
```

Comment:
Nit: slightly better wording
```suggestion
            # columns unique to the left or right hand side df
```